native/src/main/java/io/ballerina/xlibb/jsonpath/BJsonMappingProvider.java

```diff
@@ -0,0 +1,284 @@
+            }
+        } else {
+            return source;
+        }
+    }
+
+    private Object unwrapJsonValue(Object obj) {
+        if (obj == null) {
+            return null;
+        }
+        return JsonUtils.convertJSON(obj, TypeUtils.getType(obj));
+    }
+
+    private io.ballerina.runtime.api.types.Type getTargetType(Class<?> targetType) {
+        if (BMap.class.equals(targetType)) {
+            return PredefinedTypes.TYPE_MAP;
+        } else if (BArray.class.equals(targetType)) {
+            return PredefinedTypes.TYPE_JSON_ARRAY;
+        } else if (BString.class.equals(targetType)) {
+            return PredefinedTypes.TYPE_STRING;
+        } else if (String.class.equals(targetType)) {
+            return PredefinedTypes.TYPE_STRING;
+        } else if (Boolean.class.equals(targetType)) {
+            return PredefinedTypes.TYPE_BOOLEAN;
+        } else if (isJsonIntegral(targetType)) {
+            return PredefinedTypes.TYPE_INT;
+        } else if (Double.class.equals(targetType) || Float.class.equals(targetType)) {
+            return PredefinedTypes.TYPE_FLOAT;
+        } else if (BigDecimal.class.equals(targetType) ) {
+            return PredefinedTypes.TYPE_DECIMAL;
+        } else {
+            return PredefinedTypes.TYPE_ANY;
+        }
+    }
+
+    private Type getFirstTypeArgument(Type targetType) {
+        if (targetType instanceof ParameterizedType) {
+            Type[] args = ((ParameterizedType) targetType).getActualTypeArguments();
+            if (args != null && args.length > 0) {
+                if (args[0] instanceof Class) {
+                    return args[0];
+                } else if (args[0] instanceof ParameterizedType) {
+                    return args[0];
+                }
+            }
+        }
+        return null;
+    }
+
+    private Collection<Object> newCollectionOfType(Class<?> collectionType) throws MappingException {
+        if (Collection.class.isAssignableFrom(collectionType)) {
+            if (!collectionType.isInterface()) {
+                @SuppressWarnings("unchecked")
+                Collection<Object> coll = (Collection<Object>) newNoArgInstance(collectionType);
+                return coll;
+            } else if (List.class.isAssignableFrom(collectionType)) {
+                return new java.util.LinkedList<>();
+            } else if (Set.class.isAssignableFrom(collectionType)) {
+                return new java.util.LinkedHashSet<>();
+            } else if (Queue.class.isAssignableFrom(collectionType)) {
+                return new java.util.LinkedList<>();
+            }
+        }
+        String className = collectionType.getSimpleName();
+        throw new MappingException("JSON array cannot be mapped to " + className);
+    }
+
+    private Object newNoArgInstance(Class<?> targetType) throws MappingException {
+        if (targetType.isInterface()) {
+            return null;
+        } else {
```

Comment:
```suggestion
        }
```
No need to have a else part.